File: frontend/types/review.ts
export interface ReviewUser {
  id: string
  name: string
  avatar: string
}

/* Review + autor, armada desde la RPC get_reviews (ver supabase/002_get_reviews.sql) */
export interface Review {
  id: number
  user_id: string
  user: ReviewUser
  rating: number
  comment: string
  created_at: string
}

export interface UpsertReviewPayload {
  rating: number
  comment: string
}

File: backend/internal/domain/review.go
package domain

import "time"

type Review struct {
	ID        uint      `gorm:"primaryKey" json:"id"`
	UserID    uint      `gorm:"uniqueIndex;not null" json:"user_id"`
	User      User      `gorm:"foreignKey:UserID" json:"user"`
	Rating    int       `gorm:"not null" json:"rating"`
	Comment   string    `gorm:"not null" json:"comment"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}
